covid_xprize/scoring/prescriptor_scoring.py

```diff
@@ -17,28 +15,18 @@ def weight_prescriptions_by_cost(pres_df, cost_df):
 
 
 def generate_cases_and_stringency_for_prescriptions(start_date, end_date, prescription_file, costs_file):
-    # Load prescriptions
-    pres_df = pd.read_csv(prescription_file)
+    # Load the prescriptions, handling Date and regions
+    pres_df = XPrizePredictor.load_original_data(prescription_file)
 
     # Generate predictions for all prescriptions
+    predictor = XPrizePredictor()
     pred_dfs = []
     for idx in pres_df['PrescriptionIndex'].unique():
         idx_df = pres_df[pres_df['PrescriptionIndex'] == idx]
         idx_df = idx_df.drop(columns='PrescriptionIndex')  # Predictor doesn't need this
-        ip_file_path = 'prescriptions/prescription_{}.csv'.format(idx)
-        os.makedirs(os.path.dirname(ip_file_path), exist_ok=True)
-        idx_df.to_csv(ip_file_path)
-        preds_file_path = 'predictions/predictions_{}.csv'.format(idx)
-        os.makedirs(os.path.dirname(preds_file_path), exist_ok=True)
-
-        # Run predictor
-        predict(start_date, end_date, ip_file_path, preds_file_path)
-
-        # Collect predictions
-        pred_df = pd.read_csv(preds_file_path,
-                              parse_dates=['Date'],
-                              encoding="ISO-8859-1",
-                              error_bad_lines=True)
+        # Generate the predictions
+        pred_df = predictor.predict_from_df(start_date, end_date, idx_df)
```

Comment:
Predict directly from the IP DataFrame and get a DataFrame back instead of relying on the file based API.